hack/istio/multicluster/setup-keycloak.sh

```diff
@@ -0,0 +1,197 @@
+TOKEN_KEY=$(curl -k -X POST https://"${KEYCLOAK_HOSTNAME}"/realms/master/protocol/openid-connect/token \
+            -d grant_type=password \
+            -d client_id=admin-cli \
+            -d username="${ADMIN_USERNAME}" \
+            -d password="${ADMIN_PASSWORD}" \
+            -d scope=openid \
+            -d response_type=id_token | jq -r '.access_token')
+
+# Replace the redirect URI with both clusters' oauth servers. Create the realm.
+echo "Creating kube realm"
+jq ".clients[] |= if .clientId == \"kube\" then .redirectUris = [\"${CLUSTER1_OPENSHIFT_OAUTH_ROUTE}/*\", \"${CLUSTER2_OPENSHIFT_OAUTH_ROUTE}/*\"] else . end" < "${SCRIPT_DIR}"/realm-export-template.json | curl -k -L https://"${KEYCLOAK_HOSTNAME}"/admin/realms -H "Authorization: Bearer $TOKEN_KEY" -H "Content-Type: application/json" -X POST -d @-
+
+# Create the kiali user
+curl -k -L https://"${KEYCLOAK_HOSTNAME}"/admin/realms/kube/users -H "Authorization: Bearer $TOKEN_KEY" -d "{\"username\": \"kiali\", \"enabled\": true, \"credentials\": [{\"type\": \"password\", \"value\": \"${KIALI_USER_PASSWORD}\"}]}" -H 'Content-Type: application/json'
+
+# Get the openshift router CA for the cluster that keycloak is running on.
+INGRESS_ROUTER_CA_FILE=$(mktemp)
+kubectl get secrets --context "${CLUSTER1_CONTEXT}" -n openshift-ingress-operator router-ca -o jsonpath='{.data.tls\.crt}' | base64 --decode > "${INGRESS_ROUTER_CA_FILE}"
+
+# Create the keycloak kube client secret on the clusters in openshift-config.
+kubectl --context "${CLUSTER1_CONTEXT}" apply -f - <<EOF
+apiVersion: v1
+stringData:
+  clientSecret: "${KEYCLOAK_KUBE_CLIENT_SECRET}"
+kind: Secret
+metadata:
+  name: openid-client-secret
+  namespace: openshift-config
+type: Opaque
+EOF
+
+kubectl --context "${CLUSTER2_CONTEXT}" apply -f - <<EOF
+apiVersion: v1
+stringData:
+  clientSecret: "${KEYCLOAK_KUBE_CLIENT_SECRET}"
+kind: Secret
+metadata:
+  name: openid-client-secret
+  namespace: openshift-config
+type: Opaque
+EOF
+
+KEYCLOAK_ISSUER_URI="https://keycloak-keycloak.${APPS_DOMAIN}"
+
+# Create the configmap with the CA in it for the west cluster.
+kubectl --context "${CLUSTER2_CONTEXT}" create configmap keycloak-oidc-client-ca-cert --from-file=ca.crt="${INGRESS_ROUTER_CA_FILE}" -n openshift-config
+
+function update_cluster_idp {
+  local cluster_context=$1
+  # Patch the OAuth config with the keycloak idp for the west cluster.
+  if [ -z "$(kubectl --context "${cluster_context}" get oauths cluster -o jsonpath='{.spec.identityProviders[?(@.name == "openid")]}')" ]; then
+    echo "openid provider doesn't exist on the cluster's oauth config. Adding it..."
+    # Need to check if the spec.identityProviders exists and if it doesn't then ensure it does
+    # before attempting to add an item to it with the patch below otherwise the patch will fail.
+    if [ -z "$(kubectl --context "${cluster_context}" get oauths cluster -o jsonpath='{.spec.identityProviders}' --context west)" ]; then
```

Comment:
I'm getting this error spit out a couple times (note, my context name is NOT "west"):
```
Error in configuration: context was not found for specified context: west
```

I think its due to `west` context name being hardcoded when it should be the context name passed in:

```suggestion
    if [ -z "$(kubectl --context "${cluster_context}" get oauths cluster -o jsonpath='{.spec.identityProviders}' --context ${cluster_context})" ]; then
```

Reply:
wait a minute, I just noticed that `--context` is passed in twice - the last one "winning". Just remove that last one:
```suggestion
    if [ -z "$(kubectl --context "${cluster_context}" get oauths cluster -o jsonpath='{.spec.identityProviders}')" ]; then
```